### PR TITLE
Mapbox styles

### LINKED ISSUE #31: Add option to choose Mapbox theme
I haven't looked at the API for Mapbox in a few months, but if I recall, it was just another param to pass. I'd be interesting if can retrieve the available to options and have a select field pre-populated.

I will look at that tonight.

### PR DESCRIPTION
**Pull request by miklb**
Adds field for Mapbox user and dropdown select of all available styles for that user for displaying Mapbox api maps. Fixes #31

### CODE REVIEW

**Comment by miklb** [comment]
Sending this pull request mostly to get feedback on what/how can be improved. Working in local test environment.

One note, to get back custom styles, an auth_token with the scope `styles:list` must be used. Otherwise the default styles available to all authorized requests is still shown.

File: includes/class-loc-config.php [hunk 1]
```diff
@@ -200,7 +236,37 @@ public static function map_provider_callback(array $args) {
...
+		$request = wp_remote_get( $url );
+
+		if ( is_wp_error( $request ) ) {
+			return false; // Bail early.
+		}
+		$body = wp_remote_retrieve_body( $request );
+
+		$data = json_decode( $body );
+
+		echo '<select name="' . $name . '">';
```
Comment (dshanske) [inline includes/class-loc-config.php]: The styles should be separated from the style generating code. Probably by a function that takes an array of values and turns it into an options list. That will allow it to be used for styles in any map provider. The code to retrieve an array of styles should then move into the map provider class

**Comment by dshanske** [comment]
I think I'm going to rewrite the API key functionality to only show the API key box of the currently selected service.

File: includes/class-loc-config.php (hunk 1 above)
Comment (miklb) [inline includes/class-loc-config.php]: > The styles should be separated from the style generating code

I don't follow. Are you saying you would want one master list of styles from all providers vs individual providers and their available styles?

**Comment by dshanske** [comment]
I think I'll try to do some work on that.

**Pull request merged**